src/data_provider/src/network/networkSolarisWrapper.hpp

```diff
@@ -0,0 +1,369 @@
+
+            return statistic;
+        }
+
+        std::string type() const override
+        {
+            const auto buffer { Utils::exec("dladm show-phys " + this->name(), 512) };
+            constexpr auto COLUMN_TYPE_INTERFACE { "MEDIA" };
+            std::string type { "" };
+
+            if (!buffer.empty() && (buffer.find("unknown subcommand") == std::string::npos))
+            {
+                auto lines { Utils::split(buffer, '\n') };
+                std::vector<std::string> headers = { };
+                std::vector<std::string> values = { };
+
+                for (auto line : lines)
+                {
+                    Utils::replaceAll(line, "\t", "");
+                    Utils::replaceAll(line, "  ", " ");
+
+                    if (headers.size() == 0)
+                    {
+                        headers = Utils::split(line, ' ');
+                    }
+                    else
+                    {
+                        values = Utils::split(line, ' ');
+                    }
+                }
+
+                try
+                {
+                    const auto it = std::find(headers.begin(), headers.end(), COLUMN_TYPE_INTERFACE);
+
+                    if (it != headers.end() && values.size() > static_cast<size_t>(it - headers.begin()))
+                    {
+                        type = values.at(it - headers.begin());
+                    }
+                }
+                catch(...)
+                {
+                }
+            }
+
+            return type;
+        }
+
+        std::string state() const override
+        {
+            return m_interfaceFlags & IFF_UP ? "up" : "down";
+        }
+
+        std::string MAC() const override
+        {
+            std::string mac { UNKNOWN_VALUE };
+            const auto buffer { Utils::exec("ifconfig " + this->name()) };
+            if (!buffer.empty())
+            {
+                const auto lines { Utils::split(buffer, '\n') };
+                for (auto line : lines)
+                {
+                    Utils::replaceAll(line, "\t", "");
+                    const auto fields { Utils::split(line, ' ') };
+                    if (fields.size() == MAC_SIZE_FIELDS && fields.front().compare("ether") == 0)
+                    {
+                        const auto components { Utils::split(fields[MAC_ADDRESS], ':') };
+
+                        for (const auto& item : components)
+                        {
+                            if (1 == item.length())
+                            {
+                                mac.append("0" + item);
+                            }
+                            else
+                            {
+                                mac.append(item);
+                            }
+
+                            if (item.compare(components.back()))
+                            {
+                                mac.append(":");
+                            }
+                        }
```

Comment:
consider this change to be more cpp friendly.
```suggestion
                        auto components { Utils::split(fields[MAC_ADDRESS], ':') };
                        std::stringstream ret;
                        ret << std::hex << std::uppercase << std::setfill('0');
                        
                        for (auto& item : components)
                        {
                            // if you want to convert hex values ​​to uppercase
                            // std::transform(item.begin(), item.end(), item.begin(), ::toupper);
                            ret << std::setw(2) << item;
                            if (&item != &strings.back())
                            {
                                ret << ":";
                            }
                        }
                        mac = ret.str();   
```

EX: https://godbolt.org/z/qbK3T3KKa